nucliadb_core/src/query_language/mod.rs

```diff
@@ -186,41 +186,78 @@ fn split_mixed(expression: BooleanExpression, context: &QueryContext) -> NodeRes
 
 #[derive(Default)]
 pub struct QueryAnalysis {
-    pub prefilter_query: Option<BooleanExpression>,
+    pub labels_prefilter_query: Option<BooleanExpression>,
+    pub keywords_prefilter_query: Option<BooleanExpression>,
     pub search_query: Option<BooleanExpression>,
 }
 
-pub fn translate(query: &str, context: &QueryContext) -> NodeResult<QueryAnalysis> {
-    let Json::Object(subexpressions) = serde_json::from_str(query)? else {
-        #[rustfmt::skip] return Err(node_error!(
-            "Unexpected query format {query}"
-        ));
-    };
-
-    let mut prefilter_queries = Vec::new();
+pub fn translate(
+    labels_query: Option<&str>,
+    keywords_query: Option<&str>,
```

Comment:
Why are you passing this parameters directly instead of inside `QueryContext`?

Reply:
I didn't want to refactor what was already done